### PR TITLE
Refactor IsHandgun to a class variable called can_holster

### PR DESCRIPTION
**Pull request by variableundefined**
## What Does This PR Do
This PR is inspired by the previous PR that made the energy sniper (Which inherit from gun/energy, which is a holsterable type) non-holsterable. 

This refactors the IsHandgun() proc, which is only used ever to check whether something is holsterable, into a class level variable called can_holster, which is checked by the can_holster proc to see if it can in fact, be holstered.

This is much more accurate to how the proc is actually used. We've never used the IsHandgun() proc anywhere to distinguish between a handgun vs rifle form factor and it never mattered in more than one case. 

Ideally the holster should be checking for more than that, but that's for another day 

## Changelog
:cl:
tweak: Code refactor behind the gun holstering code. Report any issues.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -56,6 +56,8 @@
 	var/knife_x_offset = 0
 	var/knife_y_offset = 0
 
+	var/can_holster = TRUE // Can it be holstered?
```
Comment (farie82) [inline code/modules/projectiles/gun.dm:59]: Can you make this in the autodoc format?
```
/// comment here
var/variable
```

**Review by farie82 — CHANGES_REQUESTED** [review]

**Comment by variableundefined** [comment]
Sure, also need to amend a commit because I messed up the name of the commit anyway

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (FlattestGuitar) [reply code/modules/projectiles/gun.dm:59]: Don't do it. This doesn't need documentation.

Also remove that comment, as it does not improve the readability. The name clearly conveys what this variable is for.

**Comment by variableundefined** [comment]
There we go

**Review by SteelSlayer — APPROVED** [review]

**Review by farie82 — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**